File: demos/android/MASVS-STORAGE/MASTG-DEMO-0x04/MastgTest.kt
package org.owasp.mastestapp

import android.content.Context

class MastgTest(private val context: Context) {

    companion object {
        // INSECURE: third-party API key embedded in the app package.
        private const val MAPS_API_KEY = "AIzaSyDFakeMastgDemoKeyNotARealKey12345"

        // INSECURE: cloud provider credential embedded in the app package.
        private const val AWS_ACCESS_KEY_ID = "AKIAIOSFODNN7EXAMPLE"
    }

    fun mastgTest(): String {
        // INSECURE: secret assigned to a credential-named local variable.
        val clientSecret = "s3cr3t-not-a-real-value-9f2b"

        // INSECURE: the key is also duplicated in res/values/strings.xml,
        // where it survives even if the code is obfuscated.
        val resourceKey = context.getString(R.string.maps_api_key)

        val endpoint = "https://api.example.com/v1/report"

        val sb = StringBuilder()
        sb.appendLine("Endpoint: $endpoint")
        sb.appendLine("Maps API key (code): $MAPS_API_KEY")
        sb.appendLine("Maps API key (resources): $resourceKey")
        sb.appendLine("AWS access key id: $AWS_ACCESS_KEY_ID")
        sb.appendLine("Client secret: $clientSecret")
        return sb.toString()
    }
}

File: demos/ios/MASVS-STORAGE/MASTG-DEMO-0x05/MastgTest.swift
// SUMMARY: This sample embeds third-party credentials as Swift string literals. String literals
// end up in the __TEXT.__cstring section of the compiled binary, so anyone who obtains the IPA
// can recover them without running the app, without a jailbreak, and without any tooling beyond
// a strings dump.

import Foundation

struct MastgTest {
    // INSECURE: third-party API key embedded in the app binary.
    static let mapsAPIKey = "AIzaSyDFakeMastgDemoKeyNotARealKey12345"

    // INSECURE: cloud provider credential embedded in the app binary.
    static let awsAccessKeyId = "AKIAIOSFODNN7EXAMPLE"

    // INSECURE: secret assigned to a credential-named constant.
    static let clientSecret = "s3cr3t-not-a-real-value-9f2b"

    static let endpoint = "https://api.example.com/v1/report"

    static func mastgTest(completion: @escaping (String) -> Void) {
        let value = """
            Endpoint          : \(endpoint)
            Maps API key      : \(mapsAPIKey)
            AWS access key id : \(awsAccessKeyId)
            Client secret     : \(clientSecret)

            All three credentials above are compiled into the app binary as string literals.
            """
        completion(value)
    }
}
